scrape/scrape.go

```diff
@@ -714,11 +727,18 @@ func (s *targetScraper) scrape(ctx context.Context, w io.Writer) (string, error)
 		return "", errors.Errorf("server returned HTTP status %s", resp.Status)
 	}
 
+	if s.bodySizeLimit <= 0 {
+		s.bodySizeLimit = math.MaxInt64
+	}
 	if resp.Header.Get("Content-Encoding") != "gzip" {
-		_, err = io.Copy(w, resp.Body)
+		n, err := io.Copy(w, io.LimitReader(resp.Body, s.bodySizeLimit))
```

Comment:
I am wondering what is the impact of calling the LimitReader when no limit is set.

```
func (l *LimitedReader) Read(p []byte) (n int, err error) {
     	if l.N <= 0 {
    		return 0, EOF
     	}
     	if int64(len(p)) > l.N {
     		p = p[0:l.N]
     	}
     	n, err = l.R.Read(p)
     	l.N -= int64(n)
     	return
     }
```

It seems like this could have an impact, I will run benchmarks to see if it is noticeable.

Reply:
seems no impact almostly, is it?  i can use LimitReader a `if` branch.
```
	var rspReader io.Reader = resp.Body
	if s.bodySizeLimit > 0 {
		rspReader = io.LimitReader(resp.Body, s.bodySizeLimit)
	}
        // ... 
        	if s.bodySizeLimit > 0 {
		rspReader = io.LimitReader(s.gzipr, s.bodySizeLimit)
	}
```

Reply:
I have move io.LimitReader to if branch. :)

Reply:
> Thanks. I would like to revert and come back to the version where the reader was there all the time. Benchmark did not show a difference.

revert done.